internal/juju/kubernetesClouds.go

```diff
@@ -257,6 +249,29 @@ func (c *kubernetesCloudsClient) UpdateKubernetesCloud(input UpdateKubernetesClo
 		return errors.Annotate(err, "updating kubernetes cloud")
 	}
 
+	// update the cloud credential.
```

Comment:
ah.. i see what you're trying to do here.. sneak in a new feature! can we please do this in a separate PR as it is unrelated to the no-service-account flag?